primitives/runtime/src/lib.rs

```diff
@@ -181,18 +181,39 @@ impl From<ed25519::Signature> for MultiSignature {
 	}
 }
 
+impl TryFrom<MultiSignature> for ed25519::Signature {
```

Comment:
where are these used?

Reply:
I believe this is where the requirement comes from:
https://github.com/paritytech/substrate/blob/27f900ecad6f8b8851367d5382c2213920f01e8a/frame/system/src/offchain.rs#L330